From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Hamilton <adhamilt@gmail.com>
Date: Fri, 22 Aug 2025 22:43:17 -0500
Subject: [PATCH] po: Update Translations to Build with Gettext 0.26

Gettext 0.26 validates format strings. In some cases before
the GRUB build process was converting newlines sequences (\n)
to (\<translated character>) which is invalid. Update the impacted
language SED script files to ensure newlines use the correct escape
sequence.

This avoids build errors such as:
de@hebrew.po:8192: 'msgstr' is not a valid Shell printf format string,
unlike 'msgid'. Reason: This escape sequence is invalid.

Fixes: https://savannah.gnu.org/bugs/?67353

Signed-off-by: Andrew Hamilton <adhamilt@gmail.com>
---
 po/arabic.sed   | 2 ++
 po/cyrillic.sed | 2 ++
 po/greek.sed    | 2 ++
 po/hebrew.sed   | 2 ++
 4 files changed, 8 insertions(+)

diff --git a/po/arabic.sed b/po/arabic.sed
index 3fbee7248..50bede0f8 100644
--- a/po/arabic.sed
+++ b/po/arabic.sed
@@ -81,3 +81,5 @@ s,%\([0-9]*\)زو,%\1zu,g
 s,%\([0-9]*\)كس,%\1x,g
 s,%\([0-9]*\)لكس,%\1lx,g
 s,%\([0-9]*\)للكس,%\1llx,g
+
+s,\\ن,\\n,g
diff --git a/po/cyrillic.sed b/po/cyrillic.sed
index 472f09529..d3db38838 100644
--- a/po/cyrillic.sed
+++ b/po/cyrillic.sed
@@ -104,3 +104,5 @@ s,%\([0-9]*\)зу,%\1zu,g
 s,%\([0-9]*\)ѯ,%\1x,g
 s,%\([0-9]*\)лѯ,%\1lx,g
 s,%\([0-9]*\)ллѯ,%\1llx,g
+
+s,\\н,\\n,g
diff --git a/po/greek.sed b/po/greek.sed
index 0e81625fb..1ace5fcd2 100644
--- a/po/greek.sed
+++ b/po/greek.sed
@@ -106,3 +106,5 @@ s,%\([0-9]*\)ζυ,%\1zu,g
 s,%\([0-9]*\)ξ,%\1x,g
 s,%\([0-9]*\)λξ,%\1lx,g
 s,%\([0-9]*\)λλξ,%\1llx,g
+
+s,\\ν,\\n,g
diff --git a/po/hebrew.sed b/po/hebrew.sed
index 33174bbdc..ce59e576e 100644
--- a/po/hebrew.sed
+++ b/po/hebrew.sed
@@ -89,3 +89,5 @@ s,%\([0-9]*\)זוּ,%\1zu,g
 s,%\([0-9]*\)כּס,%\1x,g
 s,%\([0-9]*\)לכּס,%\1lx,g
 s,%\([0-9]*\)ללכּס,%\1llx,g
+
+s,\\נ,\\n,g
-- 
2.51.0

